packages/vector_graphics_compiler/lib/src/svg/node.dart

```diff
@@ -145,96 +114,101 @@ class ParentNode extends PaintingNode {
   /// If `clips` is empty, the child is directly appended. Otherwise, a
   /// [ClipNode] is inserted.
   void addChild(
-    Node child, {
-    List<Path> clips = const <Path>[],
-    Node? mask,
+    AttributedNode child, {
+    String? clipId,
+    String? maskId,
+    required Resolver<List<Path>> clipResolver,
+    required Resolver<AttributedNode> maskResolver,
   }) {
-    if (clips.isNotEmpty) {
-      child = ClipNode(clips: clips, child: child);
+    Node wrappedChild = child;
+    if (clipId != null) {
+      wrappedChild = ClipNode(
+        resolver: clipResolver,
+        clipId: clipId,
+        child: wrappedChild,
+      );
     }
-    if (mask != null) {
-      child = MaskNode(mask: mask, child: child);
+    if (maskId != null) {
+      wrappedChild = MaskNode(
+        resolver: maskResolver,
+        maskId: maskId,
+        child: wrappedChild,
+        blendMode: child.attributes.blendMode,
+      );
     }
-    _children.add(child);
+    _children.add(wrappedChild);
   }
 
   @override
   AffineMatrix concatTransform(AffineMatrix currentTransform) {
-    if (transform == null) {
+    if (transform == AffineMatrix.identity) {
       return currentTransform;
     }
-    return currentTransform.multiplied(transform!);
+    return currentTransform.multiplied(transform);
   }
 
   @override
-  Node adoptPaint(Paint? newPaint) {
+  AttributedNode applyAttributes(SvgAttributes newAttributes) {
     return ParentNode(
-      id: id,
-      transform: transform,
-      color: color,
-      paint: newPaint?.applyParent(paint),
+      newAttributes.applyParent(attributes),
+      precalculatedTransform: transform,
     ).._children.addAll(_children);
   }
 
-  // Whether or not a save layer should be inserted at this node.
-  bool _requiresSaveLayer() {
-    final Paint? localPaint = paint;
-    if (localPaint == null) {
-      return false;
+  Paint? _createLayerPaint() {
+    if (attributes.blendMode != null) {
```

Comment:
So how do we determine whether or not we need a layer - just if there is a blend mode?

Reply:
There are other cases like masks and clips applied to multiple children. I think I'm missing something about group opacity but my test case for that seems to be passing...

Reply:
My test case is _not_ passing lol.

Reply:
Fixed and updated/added test